File: greentech-frontend/src/app/services/tipo-solo.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { TipoSolo } from '../models/tipo-solo.model';

@Injectable({
  providedIn: 'root'
})
export class TipoSoloService {

  private baseURL: string =  'http://localhost:8080';

  constructor(private http: HttpClient) { }

  findAll(): Observable<TipoSolo[]> {
    return this.http.get<TipoSolo[]>(`${this.baseURL}/solos`);
  }

  findAllPaginado(pagina: number, tamanhoPagina: number): Observable<TipoSolo[]> {
    const params = {
      page: pagina.toString(),
      pageSize: tamanhoPagina.toString()
    }
    return this.http.get<TipoSolo[]>(`${this.baseURL}/solos/paginado`, {params});
  }

  findById(id: string): Observable<TipoSolo> {
    return this.http.get<TipoSolo>(`${this.baseURL}/solos/${id}`);
  }

  save(tipoSolo: TipoSolo): Observable<TipoSolo> {
    return this.http.post<TipoSolo>(`${this.baseURL}/solos`, tipoSolo);
  }

  update(tipoSolo: TipoSolo): Observable<TipoSolo> {
    return this.http.put<TipoSolo>(`${this.baseURL}/solos/${tipoSolo.id}`, tipoSolo);
  }

  delete(tipoSolo: TipoSolo): Observable<any> {
    return this.http.delete<any>(`${this.baseURL}/solos/${tipoSolo.id}`);
  }

  findByNome(nome: string, pagina: number, tamanhoPagina: number): Observable<TipoSolo[]> {
    const params = {
      page: pagina.toString(),
      pageSize: tamanhoPagina.toString()
    }
    return this.http.get<TipoSolo[]>(`${this.baseURL}/solos/search/${nome}`, {params});
  }

  count(): Observable<number> {
    return this.http.get<number>(`${this.baseURL}/solos/count`);
  }

  countByNome(nome: string): Observable<number> {
    return this.http.get<number>(`${this.baseURL}/solos/search/${nome}/count`);
  }
}

File: greentech-backend/src/main/java/br/unitins/greentech/service/TipoSoloService.java
package br.unitins.greentech.service;

import java.util.List;

import br.unitins.greentech.dto.TipoSoloDTO;
import br.unitins.greentech.model.TipoSolo;

public interface TipoSoloService {
    
    List<TipoSolo> getAll();
    
    List<TipoSolo> getAll(int page, int pageSize);
    
    TipoSolo getById(Long id);

    TipoSolo insert(TipoSoloDTO tipoSoloDTO);

    TipoSolo update(Long id, TipoSoloDTO tipoSoloDTO);

    void delete(Long id);

    Long count();

    Long countByNome(String nome);

    List<TipoSolo> getByNome(String nome, int page, int pageSize);
}
